cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -25,6 +30,71 @@
 log = logging.getLogger('edx.modulestore')
 
 
+class AboutInfo(object):
+    """ About info structure to contain
+       1) Property name to use
+       2) Where to add in the index (using flags above)
+       3) Where to source the properties value
+    """
+    # Bitwise Flags for where to index the information
+    #
+    # ANALYSE - states that the property text contains content that we wish to be able to find matched within
+    #   e.g. "joe" should yield a result for "I'd like to drink a cup of joe"
+    #
+    # PROPERTY - states that the property text should be a property of the indexed document, to be returned with the
+    # results: search matches will only be made on exact string matches
+    #   e.g. "joe" will only match on "joe"
+    #
+    # We are using bitwise flags because one may want to add the property to EITHER or BOTH parts of the index
+    #   e.g. university name is desired to be analysed, so that a search on "Oxford" will match
+    #   property values "University of Oxford" and "Oxford Brookes University",
+    #   but it is also a useful property, because within a (future) filtered search a user
+    #   may have chosen to filter courses from "University of Oxford"
+    #
+    # see https://wiki.python.org/moin/BitwiseOperators for information about bitwise shift operator used below
+    #
+    ANALYSE = 1 << 0  # Add the information to the analysed content of the index
+    PROPERTY = 1 << 1  # Add the information as a property of the object being indexed (not analysed)
+
+    # Source location options - this is the function that returns the value of the property, signature should be
+    #   course, attribute_name, modulestore
+    @staticmethod
+    def fetch_from_about(course, attribute_name, modulestore):
+        """ Get about attribute from modulestore """
+        usage_key = course.id.make_usage_key('about', attribute_name)
+        try:
+            value = modulestore.get_item(usage_key).data
```

Comment:
@martynjames a few things to look into:
* if you're obtaining the `course` via [get_course()](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/mixed.py#L317) and pass `depth=None` to that call, it may prefetch much or all of the "about" data for you.  In fact `depth=1` may be faster and have the same effect, iff 'about' items are always direct children of the course object.
* a single call to [get_items()](https://github.com/edx/edx-platform/blob/master/common/lib/xmodule/xmodule/modulestore/mixed.py#L237) might be able to pick up all of the 'about' items in one shot; however you'll have to change your logic a bit in order to fetch those in advance and then look up those values in memory somewhere.
* wrapping all the activity in the `modulestore.bulk_operations` context manager may get you some efficiencies.  I'm not sure if this has the same gains for both old mongo and split mongo; I'd be more concerned about the performance of the former.